Bottom: efa63c930a1f5a9e7a7115a84a20d367a88a3eee
Top:    10cb6b8ee6d2bb001b28f1166545fe4431d7f6e1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-25 03:22:04 +0100

TEMP HACK TO INCLUDE GL HEADER FOR gui-output-events.c


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index c12250d..0369e02 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -48,6 +48,20 @@
 #include "search.h"
 #include "rats.h"
 
+/* TEMPORARY HACK WHILST THERE IS SOME GL STUFF IN THE WRONG PLACE! */
+/* The Linux OpenGL ABI 1.0 spec requires that we define
+ * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
+ * in order to get prototypes:
+ *   http://www.opengl.org/registry/ABI/
+ */
+#ifdef ENABLE_GL
+#  define GL_GLEXT_PROTOTYPES 1
+#  include <GL/gl.h>
+#  include <gtk/gtkgl.h>
+#  include "hid/common/hidgl.h"
+#endif
+/* END TEMPORARY HACK WHILST THERE IS SOME GL STUFF IN THE WRONG PLACE! */
+
 #ifdef HAVE_LIBDMALLOC
 #include <dmalloc.h>
 #endif
